src/components/fx/calc.js

```diff
@@ -10,9 +10,33 @@
 
 var Lib = require('../../lib');
 var Registry = require('../../registry');
+var baseAttrs = require('../../plots/attributes');
 
 module.exports = function calc(gd) {
     var calcdata = gd.calcdata;
+    var fullLayout = gd._fullLayout;
+
+    function makeCoerceHoverInfo(fullTrace) {
+        var moduleAttrs = fullTrace._module.attributes;
+        var attrs = moduleAttrs.hoverinfo ?
+            {hoverinfo: moduleAttrs.hoverinfo} :
+            baseAttrs;
+        var valObj = attrs.hoverinfo;
+        var dflt;
+
+        if(fullLayout._dataLength === 1) {
+            var flags = valObj.dflt === 'all' ?
+                valObj.flags.slice() :
+                valObj.dflt.split('+');
+
+            flags.splice(flags.indexOf('name'), 1);
+            dflt = flags.join('+');
+        }
+
+        return function(val) {
+            return Lib.coerce({hoverinfo: val}, {}, attrs, 'hoverinfo', dflt);
```

Comment:
I think as you have it is probably correct - an invalid entry gets the default behavior, which is `'all'` or all but name. This could come in handy, in as far as the default is really what you want most of the time, for tweaking `hoverinfo` of just a few points by providing those in the array and leaving the rest of the array `undefined` or `null`.